### PR TITLE
Fix checkstyle and Eclipse warnings

### PR DESCRIPTION
**Pull request by rouazana**

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[11d6c5d0a6d6d1d1b7e3b05faae46e60d49a94a2] [ERROR]   MailRepositoryContract.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult:475 » Execution
[11d6c5d0a6d6d1d1b7e3b05faae46e60d49a94a2] [ERROR]   MailRepositoryContract.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult:475 » Execution
```
test this please

File: third-party/linshare/src/test/java/org/apache/james/linshare/client/LinshareAPITest.java [hunk 1]
```diff
@@ -255,7 +255,8 @@ private File templateFile() throws Exception {
         return Files.createTempFile("linshare-api-test", ".temp").toFile();
     }
 
-    private List<Document> sharedDocs(List<ReceivedShare>...shares) {
+    @SafeVarargs
+    private final List<Document> sharedDocs(List<ReceivedShare>...shares) {
```
Comment (chibenwa) [inline third-party/linshare/src/test/java/org/apache/james/linshare/client/LinshareAPITest.java]: We are missing a space after `...` no?

Comment (rouazana) [reply third-party/linshare/src/test/java/org/apache/james/linshare/client/LinshareAPITest.java]: not detected by checkstyle, but fixed anyway

**Comment by rouazana** [comment]
test this please (github abuse)

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
Merge conflict

**Comment by chibenwa** [comment]
Merged

**Pull request merged**